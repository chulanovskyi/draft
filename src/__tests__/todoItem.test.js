import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import TodoItem from '../components/todoList/todoItem/todoItem';

Enzyme.configure({ adapter: new Adapter() });

console.error = message => {
  throw new Error(message);
};

describe('<TodoItem />', () => {
  let task;

  const initialProps = {
    name: 'read book',
    isActive: true,
    id: 0,
    history: [{
      changedAt: "2017-10-19T12:39:10.863Z",
      prop: "name",
      from: "",
      to: "read book"
    }],
  };

  beforeEach(() => {
    task = {...initialProps}
  });

  it('should match its snapshot', () => {
    const tree = renderer.create(<TodoItem task={task}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a `.todoList__item`', () => {
    const wrapper = shallow(<TodoItem task={task}/>);
    expect(wrapper.find('.todoList__item')).toHaveLength(1);
  });

  it('changes state on editing task name', () => {
    const wrapper = mount(<TodoItem task={task}/>);
    wrapper.find('.item__editTask').simulate('click');
    const inp = wrapper.find('.item__editInput');
    inp.simulate('keypress', {key: 's'});
    expect(wrapper.state('editedName')).toBe('read books');
  });

  // it('simulates open modal on todoItem remove', () => {
  //   const onButtonClick = sinon.spy();
  //   const wrapper = shallow(<TodoItem task={task}/>);
  //   wrapper.find('.item__removeTask').simulate('click');
  //   expect(wrapper.find('.modal__header')).toHaveLength(1);
  // });
});
