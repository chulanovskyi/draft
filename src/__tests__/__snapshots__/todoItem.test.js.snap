// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> should match its snapshot 1`] = `
<li
  className="todoList__item"
>
  <div
    className="item__nameContainer"
  >
    <i
      className="fa fa-circle-o nameContainer__status"
      onClick={[Function]}
    />
    <span
      className="nameContainer__name"
      onClick={[Function]}
      title="read book"
    >
      read book
    </span>
    <i
      className="fa fa-pencil item__editTask"
      onClick={[Function]}
    />
  </div>
  <i
    className="fa fa-times item__removeTask"
    onClick={[Function]}
  />
  <div
    className="hidden"
  >
    <div
      className="history__change"
    >
      <i
        className="fa fa-times change__remove"
        onClick={[Function]}
      />
      <div
        className="change__header"
      >
        <span
          className="change__prop"
        >
          Text
        </span>
        <span
          className="change__date"
        >
          2017 M10 19 15:39:10
        </span>
      </div>
      <div
        className="change__body"
      >
        <span
          className="change__from"
          title=""
        >
          
        </span>
        <i
          className="fa fa-arrow-right change__arrow"
        />
        <span
          className="change__to"
          title="read book"
        >
          read book
        </span>
      </div>
    </div>
  </div>
  <i
    className="fa fa-long-arrow-up item__closeHistory displayNone"
    onClick={[Function]}
  />
</li>
`;
